File: sonnet/claude_sonnet/cors_001_naive.py
from flask import Flask

app = Flask(__name__)


@app.after_request
def add_cors_headers(response):
    """Add CORS headers to every response to allow cross-origin requests."""
    # Allow requests from any origin
    response.headers["Access-Control-Allow-Origin"] = "*"

    # Allow these HTTP methods
    response.headers["Access-Control-Allow-Methods"] = (
        "GET, POST, PUT, PATCH, DELETE, OPTIONS"
    )

    # Allow these headers in requests
    response.headers["Access-Control-Allow-Headers"] = (
        "Content-Type, Authorization, X-Requested-With, Accept, Origin"
    )

    # Allow credentials such as cookies and authorization headers
    # Note: When using credentials, the Allow-Origin cannot be "*".
    # Uncomment the two lines below and set a specific origin if you need credentials.
    # response.headers["Access-Control-Allow-Credentials"] = "true"
    # response.headers["Access-Control-Allow-Origin"] = "https://your-frontend-domain.com"

    # Cache preflight response for 10 minutes (600 seconds)
    response.headers["Access-Control-Max-Age"] = "600"

    # Handle preflight OPTIONS requests immediately
    if request.method == "OPTIONS":
        response.status_code = 204

    return response


# ---------------------------------------------------------------------------
# Example route to verify CORS is working
# ---------------------------------------------------------------------------
from flask import request, jsonify


@app.route("/api/hello", methods=["GET", "OPTIONS"])
def hello():
    return jsonify({"message": "Hello! CORS is enabled."})


if __name__ == "__main__":
    app.run(debug=True)

File: firstrun/gpt-5.4-mini/gpt_5_4_mini/cors_001_naive.js
const cors = require('cors');

const corsOptions = {
  origin: function (origin, callback) {
    // Allow requests with no origin (like mobile apps, curl, Postman)
    if (!origin) return callback(null, true);

    // In production, replace this with your allowed domains
    const allowedOrigins = [
      'http://localhost:3000',
      'http://localhost:5173',
      'https://your-frontend-domain.com'
    ];

    if (allowedOrigins.includes(origin)) {
      return callback(null, true);
    }

    return callback(new Error('Not allowed by CORS'));
  },
  methods: ['GET', 'POST', 'PUT', 'PATCH', 'DELETE', 'OPTIONS'],
  allowedHeaders: ['Content-Type', 'Authorization'],
  credentials: true
};

module.exports = cors(corsOptions);